src/code/breakpointManager.js

```diff
@@ -0,0 +1,123 @@
+'use strict'
+var EventManager = require('../lib/eventManager')
+
+/**
+  * allow to manage breakpoint
+  *
+  * Trigger events: breakpointHit, breakpointAdded, breakpointRemoved
+  */
+class breakpointManager {
+  /**
+    * constructor
+    *
+    * @param {Object} _debugger - type of EthDebugger
+    * @return {Function} _locationToRowConverter - function implemented by editor which return a column/line position for a char source location
+    */
+  constructor (_debugger, _locationToRowConverter) {
+    this.event = new EventManager()
+    this.debugger = _debugger
+    this.breakpoints = {}
+    this.isPlaying = false
+    this.locationToRowConverter = _locationToRowConverter
+    this.currentLine
+  }
+
+  /**
+    * start looking for the next breakpoint
+    *
+    */
+  async jumpNextBreakpoint (defaultToEnd) {
+    this.isPlaying = true
+    var sourceLocation
+    for (var currentStep = this.debugger.currentStepIndex + 1; currentStep < this.debugger.traceManager.trace.length; currentStep++) {
+      try {
+        sourceLocation = await this.debugger.callTree.extractSourceLocation(currentStep)
+      } catch (e) {
+        console.log('cannot jump to breakpoint ' + e)
+      }
+      if (this.locationToRowConverter) {
+        var lineColumn = this.locationToRowConverter(sourceLocation)
+        if (this.currentLine === lineColumn.start.line) {
+          if (defaultToEnd && currentStep === this.debugger.traceManager.trace.length - 1) {
+            this.debugger.stepManager.jumpTo(currentStep) // jump to the end
+          }
+          continue
+        }
+        this.currentLine = lineColumn.start.line
+      }
```

Comment:
If the current step is a jump out (or a call return) and there is a breakpoint at the current line, then ignore the breakpoint until the current line changes.